src/render/view/components/Controls.tsx

```diff
@@ -33,7 +39,25 @@ export class Controls extends React.Component<IControlsProps, IControlsState> {
             mute: false,
         };
 
+        // Setup bindings
+        this.onVolumeButtonClick = this.onVolumeButtonClick.bind(this);
+        this.onCastButtonClick = this.onCastButtonClick.bind(this);
+        this.onFullscreenButtonClick = this.onFullscreenButtonClick.bind(this);
+        this.onReconnectClick = this.onReconnectClick.bind(this);
+        this.onPlayPauseClick = this.onPlayPauseClick.bind(this);
+        this.onVolumeChange = this.onVolumeChange.bind(this);
+        this.onPlaybackChange = this.onPlaybackChange.bind(this);
+
+        // Initialization
+        this.volumeSliderValue = 100;
         this.formattedTime = this.secondsToHms(this.state.time);
+
+        // Determine what buttons to render
+        this.renderReconnect = (this.props.onReconnectButton !== undefined);
+        this.renderCopy = (this.props.onCopyButton !== undefined);
+
+        // Generate optional buttons
+        this.generateOptionalButtons();
```

Comment:
Don't like how this was done, should refactor using appropriate conditional rendering techniques